In [1]:
import pandas as pd
import numpy as np


import matplotlib.pyplot as plt
import seaborn as sns

# Acquire

In [ ]:
df = pd.read_csv('diabetes_prediction_dataset.csv')

In [41]:
df

gender   age  hypertension  heart_disease smoking_history    bmi  \
0      Female  80.0             0              1           never  25.19   
1      Female  54.0             0              0         No Info  27.32   
2        Male  28.0             0              0           never  27.32   
3      Female  36.0             0              0         current  23.45   
4        Male  76.0             1              1         current  20.14   
...       ...   ...           ...            ...             ...    ...   
99995  Female  80.0             0              0         No Info  27.32   
99996  Female   2.0             0              0         No Info  17.37   
99997    Male  66.0             0              0          former  27.83   
99998  Female  24.0             0              0           never  35.42   
99999  Female  57.0             0              0         current  22.43   

       HbA1c_level  blood_glucose_level  diabetes  
0              6.6                  140         0  
1              6.6                   80         0  
2              5.7                  158         0  
3              5.0                  155         0  
4              4.8                  155         0  
...            ...                  ...       ...  
99995          6.2                   90         0  
99996          6.5                  100         0  
99997          5.7                  155         0  
99998          4.0                  100         0  
99999          6.6                   90         0  

[100000 rows x 9 columns]

# Prep

### Are there any missing values?

In [92]:
df.isna().sum()

gender                 0
age                    0
hypertension           0
heart_disease          0
smoking_history        0
bmi                    0
HbA1c_level            0
blood_glucose_level    0
diabetic               0
dtype: int64

- zero nulls

### Are there any duplicates?

In [56]:
df.duplicated(keep='first').sum()

3854

- 3854 duplicates 
- By dropping duplicate rows, we can ensure that the model is trained on a representative and unbiased dataset, which can lead to more accurate predictions and better performance overall.


In [60]:
df.drop_duplicates(ignore_index=True, inplace=True)

In [61]:
df.duplicated().sum()

0

### What are the data types?

In [34]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100000 entries, 0 to 99999
Data columns (total 9 columns):
 #   Column               Non-Null Count   Dtype  
---  ------               --------------   -----  
 0   gender               100000 non-null  object 
 1   age                  100000 non-null  float64
 2   hypertension         100000 non-null  int64  
 3   heart_disease        100000 non-null  int64  
 4   smoking_history      100000 non-null  object 
 5   bmi                  100000 non-null  float64
 6   HbA1c_level          100000 non-null  float64
 7   blood_glucose_level  100000 non-null  int64  
 8   diabetes             100000 non-null  int64  
dtypes: float64(3), int64(4), object(2)
memory usage: 6.9+ MB


### Shape?

In [62]:
df.shape

(96146, 9)

### Changing age to an int and not a float?

In [65]:
df.age.value_counts()

80.00    4932
51.00    1566
47.00    1511
48.00    1508
53.00    1498
         ... 
0.48       83
1.00       83
0.40       66
0.16       59
0.08       36
Name: age, Length: 102, dtype: int64

In [77]:
df[(df.age < 1) & (df.diabetes == 1)]

Empty DataFrame
Columns: [gender, age, hypertension, heart_disease, smoking_history, bmi, HbA1c_level, blood_glucose_level, diabetes]
Index: []

In [83]:
df = df.drop(df[df.age < 1].index)


In [84]:
df.shape

(95236, 9)

- dropped all rows with the age less than 1 years old. 
- no ages below 1 have diabetes so it felt unneeded to keep.
- research also shows that it is rare for a child under the age of 1 to have diabetes although not impossible.
- for the sake of my data types I'm dropping them to be able to convert my age to an int

In [87]:
df.age = df.age.astype(int)

In [88]:
df.head()

gender  age  hypertension  heart_disease smoking_history    bmi  \
0  Female   80             0              1           never  25.19   
1  Female   54             0              0         No Info  27.32   
2    Male   28             0              0           never  27.32   
3  Female   36             0              0         current  23.45   
4    Male   76             1              1         current  20.14   

   HbA1c_level  blood_glucose_level  diabetes  
0          6.6                  140         0  
1          6.6                   80         0  
2          5.7                  158         0  
3          5.0                  155         0  
4          4.8                  155         0

### What are my value counts for the target now?

In [89]:
df.diabetes.value_counts()

0    86754
1     8482
Name: diabetes, dtype: int64

### Changing target name 'diabetes' to 'diabetic'

In [90]:
df = df.rename(columns={'diabetes': 'diabetic'})

In [91]:
df

gender  age  hypertension  heart_disease smoking_history    bmi  \
0      Female   80             0              1           never  25.19   
1      Female   54             0              0         No Info  27.32   
2        Male   28             0              0           never  27.32   
3      Female   36             0              0         current  23.45   
4        Male   76             1              1         current  20.14   
...       ...  ...           ...            ...             ...    ...   
96141  Female   36             0              0         No Info  24.60   
96142  Female    2             0              0         No Info  17.37   
96143    Male   66             0              0          former  27.83   
96144  Female   24             0              0           never  35.42   
96145  Female   57             0              0         current  22.43   

       HbA1c_level  blood_glucose_level  diabetic  
0              6.6                  140         0  
1              6.6                   80         0  
2              5.7                  158         0  
3              5.0                  155         0  
4              4.8                  155         0  
...            ...                  ...       ...  
96141          4.8                  145         0  
96142          6.5                  100         0  
96143          5.7                  155         0  
96144          4.0                  100         0  
96145          6.6                   90         0  

[95236 rows x 9 columns]